### PR TITLE
refactor: remove use of 'any' type

### PR DESCRIPTION
**Pull request by shahmargi12**
## Description

- **Replaced `any` Type**: Updated the code to use specific types instead of `any`.
- **Removed ESLint Comments**: Deleted unnecessary `eslint-disable-next-line` comments that were bypassing type-related linting rules.

## Why

- Replacing `any` type with specific types improves type safety, helps catch errors early, and makes the code easier to read and maintain.
 
## Issue

[311](https://github.com/eclipse-tractusx/portal-shared-components/issues/311)

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally

### CODE REVIEW

File: src/components/basic/ImageGallery/types.ts [hunk 1]
```diff
@@ -36,6 +38,5 @@ export interface ImageType {
   modalWidth?: string
   width?: string
   height?: string
-  // eslint-disable-next-line
-  additionalStyles?: any
+  additionalStyles?: CSSProperties
```
Comment (manojava-gk) [inline src/components/basic/ImageGallery/types.ts:41]: same as above

File: src/components/basic/MultiSelectList/index.tsx [hunk 2]
```diff
@@ -53,7 +55,7 @@ export interface MultiSelectListProps
   tagSize?: TagSizeType
   filterOptionsArgs?: IHashMap<string>
   defaultValues?: []
-  onAddItem: (items: []) => void
+  onAddItem: (items: Options) => void
```
Comment (manojava-gk) [inline src/components/basic/MultiSelectList/index.tsx:58]: same as above

File: src/components/basic/MultiSelectList/index.tsx [hunk 3]
```diff
@@ -79,12 +81,10 @@ export const MultiSelectList = ({
...
   const selectHeight = popperHeight ? `${popperHeight}px` : 'auto'
   const theme = useTheme()
-  const [selected, setSelected] = useState<[]>([])
+  const [selected, setSelected] = useState<Options>([])
   const [showItems, setShowItems] = useState(false)
 
-  // Add an ESLint exception until there is a solution
-  // eslint-disable-next-line
-  const handleChange = (selectedItems: any) => {
+  const handleChange = (selectedItems: Options) => {
```
Comment (manojava-gk) [inline src/components/basic/MultiSelectList/index.tsx:87]: same as above

File: src/components/basic/SelectList/index.tsx [hunk 4]
```diff
@@ -28,20 +28,19 @@ import uniqueId from 'lodash/uniqueId'
 import isEqual from 'lodash/isEqual'
 import { useEffect, useState } from 'react'
 
+type Option = Record<string, string | number | null>
 interface SelectListProps extends Omit<TextFieldProps, 'variant'> {
-  // eslint-disable-next-line
-  items: any
+  items: Option[]
```
Comment (manojava-gk) [inline src/components/basic/SelectList/index.tsx:33]: same as above

File: src/components/basic/Table/PageLoadingTable.tsx [hunk 5]
```diff
@@ -23,11 +23,15 @@ import { Table, type TableProps } from '.'
 import { LoadMoreButton } from '../Button/LoadMoreButton'
 import { hasMorePages, getMaxRows } from './components/Helper/helper'
 
-export interface PaginFetchArgs {
+export interface PaginFetchArgs<T> {
   page: number
-  // Add an ESLint exception until there is a solution
-  // eslint-disable-next-line
-  args?: any
+  args?:
```
Comment (manojava-gk) [inline src/components/basic/Table/PageLoadingTable.tsx:28]: Same as above

**Review by manojava-gk — COMMENTED** [review]
We had tried few changes like this couple of months back(in older package version) where all issues are resolved in shared-component repo but portal app got broken because of these changes. Pls make sure that you build portal app locally with this change and do a sanity check all the pages where these components are being used.

**Comment by shahmargi12** [comment]
> We had tried few changes like this couple of months back(in older package version) where all issues are resolved in shared-component repo but portal app got broken because of these changes. Pls make sure that you build portal app locally with this change and do a sanity check all the pages where these components are being used.

@manojava-gk , Thanks for the reminder. I’ve already built the portal app locally with these changes and performed a thorough sanity check on all relevant pages. I’ve shared the details in the PR: [#1113](https://github.com/eclipse-tractusx/portal-frontend/pull/1113). Please review it and let me know if there are any additional concerns.

File: src/components/basic/MultiSelectList/index.tsx (hunk 3 above)
Comment (shahmargi12) [reply src/components/basic/MultiSelectList/index.tsx:87]: This change does not affect the portal app’s functionality

File: src/components/basic/ImageGallery/types.ts (hunk 1 above)
Comment (shahmargi12) [reply src/components/basic/ImageGallery/types.ts:41]: This change does not affect the portal app’s functionality

File: src/components/basic/SelectList/index.tsx (hunk 4 above)
Comment (shahmargi12) [reply src/components/basic/SelectList/index.tsx:33]: This change does not affect the portal app’s functionality

**Review by oyo — APPROVED** [review]

**Pull request merged**